iroha/src/lib.rs

```diff
@@ -100,20 +100,24 @@ impl Iroha {
         let last_round_time = Arc::clone(&self.last_round_time);
         self.pool.spawn_ok(async move {
             loop {
-                blockchain
-                    .lock()
-                    .await
-                    .accept(
-                        sumeragi
-                            .lock()
-                            .await
-                            .sign(queue.lock().await.pop_pending_transactions())
-                            .await
-                            .expect("Failed to sign transactions."),
-                    )
-                    .await
-                    .expect("Failed to accept transactions into blockchain.");
-                *last_round_time.lock().await = Instant::now();
+                //TODO: decide what should be the minimum and maximum number of transactions
+                // and minimum time to accumulate them before creating a block
+                if queue.lock().await.len() > 0 {
```

Comment:
We lock queue twice. What about popping pending transactions first, checking that there are more than 0 transactions and passing them to sumeragi?